File: BEECROWD/C/3104/Main.c
/* Objetivo: Ler um número, que representa uma senha, e exibir o número que vem depois dele.
 * Autor   : William Silva (https://github.com/unclWill)
 * Data    : 27/09/2023
 * Tipo    : 5 (MATEMÁTICA)
 * ID      : 3104
 * Nome    : Resto 2.0
 * Link    : https://www.beecrowd.com.br/judge/pt/problems/view/3104
 */

/*******************************************************************************
****************************** ESTE PROBLEMA AINDA *****************************
******************************  NÃO FOI RESOLVIDO. *****************************
********************************************************************************/

#include <stdio.h>
#include <string.h>

int resto_divisao(char A[], int B)
{
    int i, tamanho = strlen(A);
    int resto = 0;

    for (i = 0; i < tamanho; i++)
    {
        resto = (resto * 10 + (A[i] - '0')) % B;
    }

    return resto;
}

int main()
{
    char A[100001];
    int B;

    scanf("%s %d", A, &B);

    int resultado = resto_divisao(A, B);
    printf("%d\n", resultado);

    return 0;
}

File: BEECROWD/Java/3104/Main.java
/* Objetivo: Ler um número, que representa uma senha, e exibir o número que vem depois dele.
 * Autor   : William Silva (https://github.com/bruwno)
 * Data    : 05/10/2025
 * Tipo    : 5 (MATEMÁTICA)
 * ID      : 3104
 * Nome    : Resto 2.0
 * Link    : https://www.beecrowd.com.br/judge/pt/problems/view/3104
 */

import java.io.IOException;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);

        String a = sc.nextLine();
        int b = Integer.parseInt(sc.nextLine());
        long resto = 0;

        for (char digito : a.toCharArray()) {
            int valor = digito - '0';
            resto = (resto * 10 + valor) % b;
        }

        System.out.println(resto);
    }
}